docs/dev/how-to/send-transaction-l1-l2.md

```diff
@@ -0,0 +1,379 @@
+
+    The import gives access to the [`IZkSync.sol`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/interfaces/IZkSync.sol#L4) inherited interfaces.
+
+    Import the contracts with yarn (recommended), or [download the contracts](https://github.com/matter-labs/v2-testnet-contracts).
+
+    ::: code-tabs
+    @tab yarn
+    ```yarn
+    yarn add -D @matterlabs/zksync-contracts
+    ```
+    :::
+
+2. Send the transaction by calling the [`requestExecute`](https://github.com/matter-labs/zksync-2-dev/blob/e25e6c3909dc41e71d9f377a2aeef9ddb8f4987f/sdk/zksync-web3.js/src/adapters.ts#L553) method and passing the `contractAddress` and `calldata` as a minimum. 
+
+    :::tip 
+    - The system takes care of gas estimation implicitly when not providing gas estimation parameters.
+    :::
+
+    ::: code-tabs
+    @tab TypeScript
+    ```ts
+    async requestExecute(transaction: {
+        contractAddress: Address;
+        calldata: BytesLike;
+        l2GasLimit?: BigNumberish;
+        l2Value?: BigNumberish;
+        factoryDeps?: ethers.BytesLike[];
+        operatorTip?: BigNumberish;
+        gasPerPubdataByte?: BigNumberish;
+        refundRecipient?: Address;
+        overrides?: ethers.PayableOverrides;
+    }): Promise<PriorityOpResponse> {
+        const requestExecuteTx = await this.getRequestExecuteTx(transaction);
+        return this._providerL2().getPriorityOpResponse(await this._signerL1().sendTransaction(requestExecuteTx));
+    }
+    ```
+    :::
+
+    :::info 
+    - The `calldata` format is a bytestream containing the contract address and the method call data. 
+    - The contract ABI is required to set up the calldata for sending.
+    - See the [cross chain governance tutorial implementation](../tutorials/cross-chain-tutorial.md#call-l2-contract-from-l1) for further clarification.
+    :::
+
+### Example code
+
+TODO:
+
+## Step-by-step with gas estimation
+
+1. Import the zkSync Era library or contract containing the required functionality. 
+
+    The import gives access to the [`IZkSync.sol`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/interfaces/IZkSync.sol#L4) inherited interfaces that include the gas estimation functionality.
+
+    Import the contracts with yarn (recommended), or [download the contracts](https://github.com/matter-labs/v2-testnet-contracts) from the repo.
+
+    ::: code-tabs
+    @tab yarn
+    ```yarn
+    yarn add -D @matterlabs/zksync-contracts
+    ```
+    :::
+
+2. Get the current L1 gas price. Function available in [Yul](https://github.com/matter-labs/era-system-contracts/blob/4e0487b940fa76d26107ca725d65e7abb6d0dee8/bootloader/bootloader.yul#L1069) and then called by the [JavaScript SDK](https://github.com/matter-labs/zksync-era/blob/48fe6e27110c1fe1a438c5375fb256890e8017b1/sdk/zksync-web3.js/src/provider.ts#L265).
```

Comment:
You can not estimate gas in L1. 

If you need to get the current gas price on L1, users should use `tx.gasprice` which is built-in in Solidity. The snippet that is shown here is part of the bootloader, which is the entry point for the VM. Users should not use snippets from there as there not relevant here

Reply:
thanks, can you supply a link to the Solidity function please @StanislavBreadless ?

Reply:
@barakshani ?

Reply:
?

Reply:
All input welcome, if you have a link to the Solidity function that would be very helpful.

Reply:
@kailash-manasarovar sorry, most likely I don't understand what you mean, `tx.gasprice` is just a built-in function in Solidity, if a link is needed, it can be found with Google

Reply:
I have resolved the confusion myself. Please check.

Reply:
this should be ok now, please check and resolve @StanislavBreadless